### DISCUSSION TITLE
Should we have different `withdraw` functions for the recipient and the sender

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/evm-monorepo#716] [Ideas]
As I've mentioned in [this discussion:](https://github.com/sablier-labs/v2-periphery/discussions/179)

> I realized that if we have separate functions for the sender and for the recipient (implicitly for approved operators) in core, it would allow the recipients to use ERC721.approve and let the sender withdraw to any address for them. 

At the moment if the sender is an approved operator it would still revert because the check is made in the same function:

https://github.com/sablier-labs/v2-core/blob/91887daff5872c7fa83fe3644b6f7582e9bffc0d/src/abstracts/SablierV2Lockup.sol#L251-L262

What do you think? Should we have separate functions for withdraw? @PaulRBerg @razgraf

**Comment by PaulRBerg** [comment]
I am against introducing a mirror function just for the sake of this little functionality.

I suggest we ignore this feature and continue with building a global set of [Sablier rescuers](https://github.com/sablier-labs/v2-core/discussions/155).
